ufl/algorithms/formtransformations.py

```diff
@@ -416,7 +417,26 @@ def compute_form_action(form, coefficient):
 
     parts = [arg.part() for arg in arguments]
     if set(parts) - {None}:
-        raise ValueError("compute_form_action cannot handle parts.")
+        # We assume that for MixedFunctionSpace that the max arity can be two
+        highest_arity_form = compute_form_lhs(form)
+        if highest_arity_form == 0 or highest_arity_form.empty():
+            highest_arity_form = compute_form_rhs(form)
+        if highest_arity_form == 0 or highest_arity_form.empty():
```

Comment:
Shouldn't this `if` be within the previous one to avoid testing the same thing twice ? 
```
if highest_arity_form == 0 or highest_arity_form.empty():
    highest_arity_form = compute_form_rhs(form)
    if highest_arity_form == 0 or highest_arity_form.empty():
        raise ValueError("No arguments to replace in form.")
```

Reply:
Agreed